src/components/TransactionList.tsx

```diff
@@ -28,70 +28,108 @@ import { useAppDispatch, useAppSelector } from '@/hooks/redux'
+const TransactionList = ({
+  className,
+  addressHashes,
+  title,
+  limit,
+  compact,
+  hideHeader = false,
+  hideFromColumn = false,
+  directions,
+  assetIds
+}: TransactionListProps) => {
   const { t } = useTranslation()
   const navigate = useNavigate()
   const dispatch = useAppDispatch()
-  const address = useAppSelector((state) => selectAddressByHash(state, addressHash ?? ''))
-  const allAddresses = useAppSelector(selectAllAddresses)
-  const allAddressHashes = useAppSelector(selectAddressIds) as AddressHash[]
-  const addressHashes = addressHash ? [addressHash] : allAddressHashes
-  const addresses = addressHash && address ? [address] : allAddresses
-  const [confirmedTxs, pendingTxs, allTransactionsLoaded, isLoading] = useAppSelector((s) => [
-    selectAddressesConfirmedTransactions(s, addressHashes),
-    selectAddressesPendingTransactions(s, addressHashes),
-    s.addresses.allTransactionsLoaded,
+
+  const addresses = useAppSelector(
+    addressHashes && addressHashes.length > 0 ? (s) => selectAddresses(s, addressHashes) : selectAllAddresses
+  )
+  const hashes = addresses.map((address) => address.hash)
+  const [confirmedTxs, pendingTxs, isLoading] = useAppSelector((s) => [
+    selectAddressesConfirmedTransactions(s, hashes),
+    selectAddressesPendingTransactions(s, hashes),
     s.addresses.loading
   ])
 
   const [selectedTransaction, setSelectedTransaction] = useState<AddressConfirmedTransaction>()
+  const [nextPageToLoad, setNextPageToLoad] = useState(1)
+  const [showAllTransactionsLoadedMsg, setShowAllTransactionsLoadedMsg] = useState(false)
 
   const singleAddress = addresses.length === 1
+  const filteredConfirmedTxs = applyFilters({ txs: confirmedTxs, directions, assetIds, hideFromColumn })
+  const displayedConfirmedTxs = limit ? filteredConfirmedTxs.slice(0, limit - pendingTxs.length) : filteredConfirmedTxs
   const totalNumberOfTransactions = addresses.map((address) => address.txNumber).reduce((a, b) => a + b, 0)
-  const showSkeletonLoading = isLoading && !confirmedTxs.length && !pendingTxs.length
-  const displayedConfirmedTxs = limit ? confirmedTxs.slice(0, limit - pendingTxs.length) : confirmedTxs
-  const showAllTransactionsLoadedMsg = singleAddress ? address?.allTransactionPagesLoaded : allTransactionsLoaded
+  const showSkeletonLoading = isLoading && !filteredConfirmedTxs.length && !pendingTxs.length
+
+  // TODO: How do we handle paging when addresses filtering changes? We need to keep track of the loaded page for every
+  // combination of selected addresses and in general all filtering criteria. That sounds very complex. Is there a
+  // simpler way? What if no matter what the filtering criteria are, we always:
+  // 1. fetch the next page of ALL addresses
+  // 2. check to see if we received any results
+  // 3. filter the results
+  // 4. if the filter results are 0 but the results are not 0, fetch again
```

Comment:
Drawing your attention on this one to hear your thoughts.

Reply:
This sounds good to me - in fact, the only solution I see - until we have a more complex endpoint which would allow more query params. :)

Reply:
I'll tackle this in a follow-up PR if that's ok